nucypher/config/characters.py

```diff
@@ -322,19 +334,19 @@ def assemble(cls, filepath: str = None, **overrides) -> dict:
         return payload
 
     @classmethod
-    def generate_runtime_filepaths(cls, config_root: str) -> dict:
+    def generate_runtime_filepaths(cls, config_root: Path) -> dict:
         """Dynamically generate paths based on configuration root directory"""
         filepaths = dict(config_root=config_root,
-                         config_file_location=os.path.join(config_root, cls.generate_filename()))
+                         config_file_location=config_root / cls.generate_filename())
         return filepaths
 
-    def initialize(self, password: str = None) -> str:
+    def initialize(self, password: str = None) -> Path:
```

Comment:
`Optional`

Reply:
I think it always returns `Path`: if `self.dev==True` it assings new `self.config_root`, otherwise it throws inside `self._ensure_config_root_exists()`

Reply:
Bogdan meant for the password parameter i.e. `password: Optional[str] = None`.